### PR TITLE
Move the benchmark logic in a module

### PR DESCRIPTION
**Pull request by piuccio**
This allows to use the bench tool from other applications.

Hi I've moved everything from `bin/bench` to `lib/bench.js` except for argument parsing. I'd like to use your benchmark from another application. If you're interested I'd also like to extend the benchmark to be more modular. I'll do it in another pull request.

Would be nice to publish it on npm

### CODE REVIEW

File: lib/bench.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    results.push(row);
+  };
+
+  input.forEach(benchmark);
+
+  output(lineBreak);
+  output(results.toString());
+  output(lineBreak);
+
+};
```
Comment (jakubpawlowicz) [inline lib/bench.js]: Please make sure there's a newline at the end of the file.

**Comment by jakubpawlowicz** [comment]
Otherwise it's all good.

**Comment by piuccio** [comment]
Newline added

**Comment by jakubpawlowicz** [comment]
:+1: @piuccio

**Pull request merged**